### PR TITLE
Use scrape manager

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
This PR replaces #4 and #5 and basically we are using the prometheus scrape manager to delegate scraping work. For simplicity it's better choice for now. We can stop/start as we need if we need to do operations on the tsdb.

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -83,43 +74,77 @@ func (p *Plugin) OnActivate() error {
...
+		// we will need to push new cluster layout to p.clusterCh by either polling the cluster table
+		// or listening the cluster event messages
+		p.API.LogWarn("cluster meterics is not enabled")
 	}
 
+	// this goroutine will need to be re-structurd to listen a more channels
+	// once we start supporting HA, we will need to listen the cluster change channel and
+	// convert the []mmodel.ClusterDiscovery entries into map[string][]*targetgroup.Group
+	p.waitGroup.Add(1)
+	go func() {
```
Comment (agnivade) [inline server/plugin.go:129]: Just for clarity, you don't really need a separate goroutine at this point, but we are creating it for when we will need HA?

File: server/configuration.go [hunk 2]
```diff
@@ -222,3 +207,37 @@ func (p *Plugin) isHA() bool {
...
 }
+
+func generateTargetGroup(appCfg *model.Config, nodes []*model.ClusterDiscovery) (map[string][]*targetgroup.Group, error) {
+	host, port, err := net.SplitHostPort(*appCfg.MetricsSettings.ListenAddress)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse the listen address %q", *appCfg.MetricsSettings.ListenAddress)
+	}
+
+	sync := map[string][]*targetgroup.Group{
+		"prometheus": make([]*targetgroup.Group, 1),
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/configuration.go:218]: Ah no, it's forgotten.

File: server/plugin.go (hunk 1 above)
Comment (isacikgoz) [reply server/plugin.go:129]: Exactly, there will be `select` to listen different channels for HA. Therefore I added the comment above.

File: server/plugin.go [hunk 3]
```diff
@@ -83,43 +74,77 @@ func (p *Plugin) OnActivate() error {
...
+
+	scpCfg := &config.Config{
+		ScrapeConfigs: []*config.ScrapeConfig{
+			{
+				JobName:                    "prometheus",
+				Scheme:                     "http",
+				MetricsPath:                "metrics",
+				ScrapeInterval:             model.Duration(time.Duration(*p.configuration.ScrapeIntervalSeconds) * time.Second),
+				ScrapeTimeout:              model.Duration(time.Duration(*p.configuration.ScrapeTimeoutSeconds) * time.Second),
+				BodySizeLimit:              units.Base2Bytes(*p.configuration.BodySizeLimitBytes),
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply server/plugin.go:102]: It's already getting converted to `int64` again in the scraper, so it's just to satisfy the type.

https://github.com/prometheus/prometheus/blob/c83e1fc5748be3bd35bf0a31eb53690b412846a4/scrape/scrape.go#L270

[the remark this answers is not in the document]
Comment (agnivade) [reply server/plugin.go:102]: Ah nvm. I missed that `BodySizeLimit` is in units.Base2Bytes.

**Review by agnivade — APPROVED** [review]
Nicely done :ok_hand:

**Pull request merged**